packages/salesforcedx-visualforce-markup-language-server/src/services/htmlLinks.ts

```diff
@@ -66,37 +60,31 @@ function createLink(
     startOffset++;
     endOffset--;
   }
-  const workspaceUrl = getWorkspaceUrl(
-    documentUri,
-    tokenContent,
-    documentContext,
-    base
-  );
+  const workspaceUrl = getWorkspaceUrl(documentUri, tokenContent, documentContext, base);
   if (!workspaceUrl || !isValidURI(workspaceUrl)) {
     return null;
   }
   return {
-    range: Range.create(
-      document.positionAt(startOffset),
-      document.positionAt(endOffset)
-    ),
+    range: Range.create(document.positionAt(startOffset), document.positionAt(endOffset)),
     target: workspaceUrl
   };
 }
 
 function isValidURI(uri: string) {
   try {
-    Uri.parse(uri);
+    // reject URIs with bare % not followed by 2 hex digits
+    // I don't know if this was a requirement, but it was in the UT
+    if (uri.includes('%') && !uri.match(/%[0-9A-Fa-f]{2}/)) {
+      return false;
+    }
+    URI.parse(uri);
```

Comment:
```suggestion
    URI.parse(uri, true);
```

I went to check the [original code](https://github.com/microsoft/vscode-uri/blob/main/src/uri.ts) for the `parse()` function.  The check for `%` is already done in that function, so we don't need to add an additional check here.

The function does provide a strict mode, which defaults to **false**.  I changed it to **true** so that invalid URIs would throw an error and reach our catch block.

Reply:
I like the research, but it does cause one of our previously passing UT to fail.

Maybe the UT is invalid, but I'd rather just leave it